test/test_field.py

```diff
@@ -0,0 +1,32 @@
+import unittest
+import os.path
+
+from tableaudocumentapi import Datasource, Field
+from tableaudocumentapi.field import _find_metadata_record
+
+TEST_ASSET_DIR = os.path.join(
+    os.path.dirname(__file__),
+    'assets'
+)
+TEST_TDS_FILE = os.path.join(
+    TEST_ASSET_DIR,
+    'datasource_test.tds'
+)
+
+
+class FieldsUnitTest(unittest.TestCase):
+    def test_field_throws_if_no_data_passed_in(self):
+        try:
+            Field()
+            self.fail('Field should have thrown')
+        except AttributeError:
+            pass
+
+
+class FindMetaDataRecordEdgeTest(unittest.TestCase):
+    class MockXml(object):
```

Comment:
I personally prefer fine-grain mocks that only mock out the specific thing we are trying to mock.  If we need to resort to unittest.mock (once we drop 2.7 support), then we aren't decomposing enough for my preferences ^_^